include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -3461,6 +3514,373 @@ MGTransferBlockGlobalCoarsening<dim, VectorType>::get_matrix_free_transfer(
+    const LinearAlgebra::distributed::Vector<Number> &src) const
+{
+  std::vector<Number> evaluation_point_results;
+  std::vector<Number> buffer;
+
+  evaluation_point_results.resize(rpe.get_point_ptrs().size() - 1);
+
+  for (unsigned int j = 0; j < evaluation_point_results.size(); ++j)
+    evaluation_point_results[j] =
+      src.local_element(this->level_dof_indices_fine[j]);
+
+  // Weight operator in case some points are owned by multiple cells.
+  if (rpe.is_map_unique() == false)
+    {
+      const auto &ptr = rpe.get_point_ptrs();
+
+      for (unsigned int i = 0; i < ptr.size() - 1; ++i)
+        {
+          const auto n_entries = ptr[i + 1] - ptr[i];
+          if (n_entries == 0)
+            continue;
+
+          evaluation_point_results[i] /= n_entries;
+        }
+    }
+
+  const auto evaluation_function = [&](const auto &values,
+                                       const auto &cell_data) {
+    std::vector<Number>                    solution_values;
+    FEPointEvaluation<1, dim, dim, Number> evaluator(*mapping_info, *fe_coarse);
+
+    for (unsigned int cell = 0; cell < cell_data.cells.size(); ++cell)
+      {
+        solution_values.resize(fe_coarse->n_dofs_per_cell());
+
+        // gather and integrate
+        evaluator.reinit(cell);
+
+        for (const auto q : evaluator.quadrature_point_indices())
+          evaluator.submit_value(
+            values[q + cell_data.reference_point_ptrs[cell]], q);
+
+        evaluator.integrate(solution_values, EvaluationFlags::values);
+
+        // resolve constraints and scatter
+        internal::VectorDistributorLocalToGlobal<Number, VectorizedArrayType>
+          writer;
+        constraint_info.read_write_operation(
+          writer,
+          dst,
+          reinterpret_cast<VectorizedArrayType *>(solution_values.data()),
+          cell,
+          1,
+          solution_values.size(),
+          true);
+      }
+  };
+
+  rpe.template process_and_evaluate<Number>(evaluation_point_results,
+                                            buffer,
+                                            evaluation_function);
+}
+
+
+
+template <int dim, typename Number>
+void
+MGTwoLevelTransferNonNested<dim, LinearAlgebra::distributed::Vector<Number>>::
+  interpolate(LinearAlgebra::distributed::Vector<Number> &      dst,
+              const LinearAlgebra::distributed::Vector<Number> &src) const
+{
+  AssertThrow(false, ExcNotImplemented());
+  (void)dst;
+  (void)src;
+}
+
+
+
+template <int dim, typename Number>
+void
+MGTwoLevelTransferNonNested<dim, LinearAlgebra::distributed::Vector<Number>>::
+  enable_inplace_operations_if_possible(
+    const std::shared_ptr<const Utilities::MPI::Partitioner>
+      &external_partitioner_coarse,
+    const std::shared_ptr<const Utilities::MPI::Partitioner>
+      &external_partitioner_fine)
+{
+  this->internal_enable_inplace_operations_if_possible(
+    external_partitioner_coarse, external_partitioner_fine, constraint_info);
```

Comment:
Related to my comment above, this at least indicates that `enable_inplace_operations_if_possible` in `MGTwoLevelTransferBase` should not be pure virtual and directly expose this functionality. Otherwise, see the other comments.